.github/workflows/flakeguard.yml

```
@@ -446,7 +464,7 @@ jobs:
           fi
 
       - name: Post comment on PR if flaky tests found
-        if: ${{ fromJSON(steps.results.outputs.summary).flaky_tests > 0 && github.event_name == 'pull_request' }}
+        if: ${{ (success() || failure()) && fromJSON(steps.results.outputs.summary).flaky_tests > 0 && github.event_name == 'pull_request' }}
```

Is this different from `always()`?
```suggestion
        if: ${{ always() && fromJSON(steps.results.outputs.summary).flaky_tests > 0 && github.event_name == 'pull_request' }}
```

---

Yes, `always()` will also trigger if the workflow has been `cancelled`, which we would rather not do here. It's not a huge difference, but could stop some confusion.